### ISSUE TITLE
Vomnibar search result order is less relevant than Firefox URL bar search

### ISSUE THREAD

**Issue by Lcchy** [gdh1995/vimium-c#1137]
Hi, first of all, thank you for the great extension, I use it every day!

**Describe the bug**

This is not completely a bug, more of a sub-optimal search result order, in my experience.
When searching in Vomnibar, it returns results from history in an order that does not match the search results from the Firefox address bar.
It seems to me that Vomnibar does maybe not take into account how often a history entry was visited to return the most likely candidates, while the Firefox url bar does?

Would there be a way to take into account the history visit frequency to prioritize URLs that were visited a lot recently?
If that is already the case, could there be a way to configure the search behavior (e.g. give more weight to visit frequency) ?

**Example**

Below you can see the results from a history search of "search".
The result "search.nixos.org" ranks 4th.
When I try the same in the Firefox URL bar, this result is immediately the first, as its my most frequent use.

![screenshot_2024-May-19th_20h38m33s](https://github.com/gdh1995/vimium-c/assets/52164716/92496011-f2a1-4015-8444-a8d459a8bb05)

**Browser, OS and Vimium C versions**

- Browser name: Firefox
- Browser version: 125.0.3
- Vimium C version: 1.99.997
- OS name and version: NixOS 23.11

**Comment by gdh1995** [comment]
Yes you're right - Vimium C hasn't used visit frequency (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/history/HistoryItem#visitcount) to sort history items.

It's mainly because the original project `Vimium` (https://github.com/philc/vimium) didn't use it. What' more, I'm not sure how to use the count.

**Comment by Lcchy** [comment]
Sorry for not getting back in so long! 

I see, I have worked around this by prioritizing the domains in my Vomnibar shortcut, like this:
`map o Vomnibar.activate icase engines="domain,history"`

This gives me better results personally.

FYI: Mozilla actually pioneered the so called "frecency" sorting algorithm that is widely used.
Here is some doc about it: https://firefox-source-docs.mozilla.org/browser/urlbar/ranking.html

I completely understand if this is not in the scope of this project, feel free to close the issue if you want.
